internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -473,6 +473,77 @@ func (mc *mac) generateRouteKeys(stream *livepeer.CreateStreamResp) []string {
 	return keys
 }
 
+func (mc *mac) triggerUserNew(w http.ResponseWriter, r *http.Request, lines []string, rawRequest string) bool {
+	if len(lines) != 6 {
+		glog.Errorf("Expected 6 lines, got %d, request \n%s", len(lines), rawRequest)
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("false"))
+		return false
+	}
+
+	playbackId := lines[0]
+	stream, err := mc.lapi.GetStreamByPlaybackID(playbackId)
+	if err != nil || stream == nil {
+		glog.Errorf("Error getting stream info from Livepeer API playbackId=%s err=%v", playbackId, err)
+		w.WriteHeader(http.StatusNotFound)
+		w.Write([]byte("false"))
+		return false
+	}
+
+	userId := stream.UserID
+	userWebhooks, err := mc.lapi.GetWebhooksByUserId(userId, "user.new")
+	if err != nil {
+		glog.Errorf("Error getting webhooks for user %s err=%v", userId, err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("false"))
+		return false
+	}
+
+	if len(userWebhooks) == 0 {
+		glog.V(model.DEBUG).Infof("No user.new webhooks for user %s", userId)
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("false"))
+		return false
+	}
+
+	for _, userWebhook := range userWebhooks {
+		if userWebhook.Url == "" {
+			glog.Errorf("User webhook %s has no URL", userWebhook.ID)
+			continue
+		}
+		glog.V(model.DEBUG).Infof("Calling user.new webhook %s", userWebhook.Url)
+
+		resp, err := http.Post(userWebhook.Url, "text/plain", strings.NewReader(rawRequest))
+		if err != nil {
+			glog.Errorf("Error calling user.new webhook %s err=%v", userWebhook.Url, err)
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte("false"))
+			return false
+		}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			glog.Errorf("Error reading response body from user.new webhook %s err=%v", userWebhook.Url, err)
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte("false"))
+			return false
```

Comment:
I think we should err on the side of permissiveness here. If the webhook is experiencing errors, we should allow the playback to go through by default. This will avoid affecting our QoS if the user does something wrong with their webhooks, and maybe later we can add some flag to the webhook saying that it is "disallow on failure", in case someone ever needs/prefers it like that.

Reply:
Disagree as per other point